app/controllers/bulk_downloads_controller.rb

```diff
@@ -2,8 +2,55 @@
 class BulkDownloadsController < ApplicationController
   include BulkDownloadTypesHelper
 
+  before_action do
+    allowed_feature_required("bulk_downloads")
+  end
+
   # GET /bulk_downloads/types
   def types
     render json: BulkDownloadTypesHelper::BULK_DOWNLOAD_TYPES
   end
+
+  # POST /bulk_downloads
+  def create
+    bulk_download_params = params.permit(:download_type, sample_ids: [], params: {})
+
+    # Convert sample ids to pipeline run ids.
+    pipeline_run_ids = Sample.where(id: bulk_download_params[:sample_ids]).includes(:pipeline_runs)
+                             .select do |sample|
+                         # Check that the most recent pipeline run succeeded.
+                         pr = sample.pipeline_runs.first
+                         !pr.nil? && pr.finalized && pr.job_status == PipelineRun::STATUS_CHECKED
```

Comment:
nit: did an autoformatter do this? The indentation is hard to read

Reply:
Yeah, autoformatter. Tried to make it better.